alibi_detect/models/tensorflow/trainer.py

```diff
@@ -1,21 +1,25 @@
+from functools import partial
 import numpy as np
 import tensorflow as tf
 from typing import Callable, Tuple
 
 
-def trainer(model: tf.keras.Model,
-            loss_fn: tf.keras.losses,
-            X_train: np.ndarray,
-            y_train: np.ndarray = None,
-            optimizer: tf.keras.optimizers = tf.keras.optimizers.Adam(learning_rate=1e-3),
-            loss_fn_kwargs: dict = None,
-            preprocess_fn: Callable = None,
-            epochs: int = 20,
-            batch_size: int = 64,
-            buffer_size: int = 1024,
-            verbose: bool = True,
-            log_metric:  Tuple[str, "tf.keras.metrics"] = None,
-            callbacks: tf.keras.callbacks = None) -> None:  # TODO: incorporate callbacks + LR schedulers
+def trainer(
+        model: tf.keras.Model,
+        loss_fn: tf.keras.losses,
+        x_train: np.ndarray,
+        y_train: np.ndarray = None,
+        dataset: tf.keras.utils.Sequence = None,
+        optimizer: tf.keras.optimizers = tf.keras.optimizers.Adam(learning_rate=1e-3),
+        loss_fn_kwargs: dict = None,
+        preprocess_fn: Callable = None,
```

Comment:
Can we now rename this arg `preprocess_batch_fn` given that it acts on batches and reserve `preprocess_fn` for functions that are expected to include their own batching logic if necessary? Would also be more consistent as at the moment we have `preprocess_fn=preprocess_batch_fn` in places.

Reply:
Included this in #298